### ISSUE TITLE
Grouping Checks

### ISSUE THREAD

**Issue by arosslau** [liip/LiipMonitorBundle#113]
We have implemented some checks (some built-in - some own) and recognized a need for grouping checks.

## Our Story
We have different servers for our application like ```app```, ```cron``` etc. One of our purposes is to check php-extensions which would be differently on ```app``` and ```cron```.
Another use-case is that some checks only should be available on certain servers.

For that it would be nice to pass some group parameter to the execution of the checks:
```
app/console monitor:list --group=app

app/console monitor:health --group=app
```
_(providing a group-option should be possible in api, too)_

## How-To
I think its neccessary to extend the current config for each built-in check to have a ```group``` config:
```
liip_monitor:
    checks:
        php_extensions:
            group: default # optional
            php-apc:
                enabled: true
                group: default # optional
```

For custom check the tag has to be extended:
```
services:
    monitor.check.php_extensions:
        class: Acme\HelloBundle\Check\PhpExtensionsCheck
        arguments:
            - [ xhprof, apc, memcache ]
        tags:
            - { name: liip_monitor.check, alias: php_extensions, group: default }
```

## Summary
What do think about this feature-idea? Should we provide any PR for that?

**Comment by DrSchimke** [comment]
The groups idea could even be solved in a different way (using an optional config level):

```
liip_monitor:
    enable_controller: true
    checks:
        default:
            php_version:
                5.4.36: >=
            php_extensions:
                - redis
        another_group:
            php_version:
                5.4.37: >=
            php_extensions:
                - redis
                - ldap
```

This is a bit like the doctrine dbal connections configuration.

**Comment by kbond** [comment]
There was a group system in 1.0 but it was removed in 2.0 - I think because we couldn't figure out how best to implement it. Sounds like a good idea to add it back. I like @DrSchimke's config system. Makes having checks in multiple groups easier. The service tags would need to be able to have multiple groups too.

**Comment by arosslau** [comment]
For now the Config would look like @DrSchimke explained in hin comment above. We made the old configuration work for BC. But we don't know if it should be mentioned in any documention that someone would be able to use the "old"-way.

(Background: There will always be at leat one "default"-group; internally)

When we did this we were inspired by the way doctrine handles its dbal-connections. As you may see, you would not be able to dump the documentationreference for the "old-way" - like in doctrine ([doctrine full reference](http://symfony.com/doc/current/reference/configuration/doctrine.html) vs. [how its used in Symfony2-Standard](https://github.com/symfony/symfony-standard/blob/2.8/app/config/config.yml) - look for the doctrineblock).

So my Question:
If/Where to put the information that both ways would work?

**Comment by lsmith77** [comment]
it seems like you are working on this already .. feel free to open the PR right away and prefix the title it with ``[WIP]``

**Comment by lsmith77** [comment]
should be fixed via #123
